diff --git a/src/components/select/Select.js b/src/components/select/Select.js
index a925f0b49..40d960084 100644
--- a/src/components/select/Select.js
+++ b/src/components/select/Select.js
@@ -170,7 +170,7 @@ export default class SelectComponent extends Field {
    */
   addOption(value, label, attrs = {}, id) {
     const option = {
-      value: _.isObject(value) ? value : String(value),
+      value: _.isObject(value) ? value :  _.isNull(value) ? this.emptyValue : String(value),
       label: label
     };
 
